src/com/walmartlabs/lacinia/pedestal.clj

```diff
@@ -20,6 +20,16 @@
     :headers {}
     :body body}))
 
+
+(defn extract-query-for-type [request]
+  (case (get-in request [:headers "content-type"])
+    "application/json"
```

Comment:
We also want to have the ability to extract out the operation name (if not now, then soon).

I'd propose a multi-method, keyed on content type, that returns a map with three keys that can be merged into the request map.